### PR TITLE
Update multimodal docs with new models

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

- Update `multimodal.md` to include new models supported over the past few months.
- Add `Qwen3-Omni` announce in the qwen_moe doc, including E2E test scripts.

# Tests
Build docs for [readthedocs](https://screenshot.googleplex.com/74Kw3AZjPGBzxdW):
```
# Build:
cd docs && sphinx-build -b html . _build/html

# Serve:
python -m http.server -d _build/html
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by aireenmei — APPROVED** [review]

**Review by Rohan-Bierneni — APPROVED** [review]
LGTM! Can we also add a link in https://github.com/AI-Hypercomputer/maxtext/blob/be7748b5403dbd304057b38a3ea0a2af1e8263b8/tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md#L4 that has a link to this multimodal doc for qwen3.5 section

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ Qwen3 is a family of open-source large language models from the Qwen team at Ali
 
 -   **Qwen3-480B-A35B**
 
+-   **Qwen3-Omni-30B-A3B**
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md:12]: Can you also link the paper for qwen3-omni below with the other artifacts

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# This file is documentation for how to get started with Qwen3-Omni-30B-A3B.
+# This e2e script was tested on a v5p-8 TPU VM.
+
+# This file runs Step 1 on CPU/TPU.
+# 1. Convert the HuggingFace checkpoint (bf16) to MaxText-compatible checkpoint (bf16):
+#    Unscanned format is used here as it is better suited for decoding.
+# 2. Run multimodal decoding: text+image, and text+video+audio.
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh]: The current script is doing (1) for unscanned and parts of (2). Do you want to oboard this model to xlml for daily tests? 

If so, I would recommend to follow the approach similar to qwen3.5-35b here: https://github.com/AI-Hypercomputer/maxtext/tree/main/tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b. We split into 2 scripts: first only does chkpt conversion from hf. The second script runs the train, decode, sft tests. Note that the second script pulls converted checkpoint from gcs bucket

If not, please update this comment.

**Review by Rohan-Bierneni — COMMENTED** [review]
Thank you for the change I have left some comments

File: tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md (hunk 1 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/run_qwen_moe.md:12]: Done, added a technical report.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh (hunk 2 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh]: Thanks! Just split the scripts into two: (1) ckpt conversion and (2) two multimodal decode runs.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+python3 -m pip install torch --index-url https://download.pytorch.org/whl/cpu
+
+# Setup local HF path argument if one was provided
+HF_LOCAL_ARG=""
+if [ -n "${HF_MODEL_PATH}" ]; then
+  HF_LOCAL_ARG="hf_model_path=${HF_MODEL_PATH}"
+fi
+
+# ---
+# Step 1: Checkpoint Conversion
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh:47]: Can you also add the command to convert to scanned checkpoint below. 

If scanned checkpoitn will not be used/is not supported in the checkpoint util then no worries we can skip it

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+# bash tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh
+# ---
+
+set -ex
+
+export MODEL_NAME="${MODEL_NAME:-qwen3-omni-30b-a3b}"
+export TOKENIZER_PATH="${TOKENIZER_PATH:-Qwen/Qwen3-Omni-30B-A3B-Instruct}"
+
+# Base output path where the MaxText checkpoint from Step 1 was written.
+export BASE_OUTPUT_PATH="${BASE_OUTPUT_PATH:-gs://your-gcs-bucket/qwen3-omni-30b-a3b_maxtext_ckpt}"
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: Since this will be used by xlml for tests, the default value should output to gs://runner-maxtext-logs/$(date +%Y-%m-%d-%H-%M). 

Similar to how it is set here: https://github.com/AI-Hypercomputer/maxtext/blob/3190805a8a9162ccf5d8c0b5df759a238bf3df62/tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh#L22

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  echo "Error: HF_TOKEN environment variable is not set. Please export your Hugging Face token."
+  echo "Example: export HF_TOKEN=hf_..."
+  exit 1
+fi
+
+# Strip trailing slash from base path to avoid malformed URIs
+BASE_OUTPUT_PATH=${BASE_OUTPUT_PATH%/}
+echo "Using BASE_OUTPUT_PATH = ${BASE_OUTPUT_PATH}"
+
+UNSCANNED_CKPT_PATH=${BASE_OUTPUT_PATH}/unscanned/0/items
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: XLML will pull an already converted checkpoint from gs://maxtext-model-checkpoints.

Can you upload a converted checkpoint to this gcs bucket, and set the value here to point to that checkpoint.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+# Base output path where the MaxText checkpoint from Step 1 was written.
+export BASE_OUTPUT_PATH="${BASE_OUTPUT_PATH:-gs://your-gcs-bucket/qwen3-omni-30b-a3b_maxtext_ckpt}"
+
+if [ -z "${HF_TOKEN}" ]; then
+  echo "Error: HF_TOKEN environment variable is not set. Please export your Hugging Face token."
+  echo "Example: export HF_TOKEN=hf_..."
+  exit 1
+fi
+
+# Strip trailing slash from base path to avoid malformed URIs
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: Should be able to remove this logic if using hardcoded gcs bucket gs://runner-maxtext-logs

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+BASE_OUTPUT_PATH=${BASE_OUTPUT_PATH%/}
+echo "Using BASE_OUTPUT_PATH = ${BASE_OUTPUT_PATH}"
+
+UNSCANNED_CKPT_PATH=${BASE_OUTPUT_PATH}/unscanned/0/items
+
+# ---
+# Step 2a: Multimodal Decode — text + image
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh:38]: Before the decode tests, do you think we should have forward_pass_logit_check test as well. This would compare to hf golden logits and would need to be stored in specific gcs bucket. 

Ex: https://github.com/AI-Hypercomputer/maxtext/blob/3190805a8a9162ccf5d8c0b5df759a238bf3df62/tests/end_to_end/tpu/qwen/moe/qwen3.5-35b-a3b/2_test_qwen3.5_35b_a3b.sh#L43

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+# Uses a test image from the repo assets.
+# max_prefill_predict_length accounts for image tokens (~256) + text prompt tokens.
+# ---
+python3 -m maxtext.inference.decode src/maxtext/configs/base.yml \
```
Comment (Rohan-Bierneni) [inline tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: The way we point to base.yml file currently in toher scripts is like this:

```
${MAXTEXT_CONFIGS_DIR:-${MAXTEXT_REPO_ROOT:-$PWD}/src/maxtext/configs}/base.yml
```

Can we change it to this for other instances in this script for consistency.

**Review by Rohan-Bierneni — CHANGES_REQUESTED** [review]
The overall script are formatted very good split between part 1 and 2. Mainly we need some changes to script 2 to make it compatible with our xlml tests.

We also need another pr in xlml once this is merged to add the tests in our DAG.

Thank you for making these changes!

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh (hunk 7 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh:38]: That would be great but vision branch may undergo some precision issue, as we translated image processing steps from Torch to JAX, and some interpolation operations cannot closely match. So now we mostly look at whether the output are making sense. I will do that as a follow up.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh (hunk 3 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/1_test_qwen3_omni_30b_a3b.sh:47]: Due to the deepstack features in the vision encoder, Omni doesn't have a full scanned support. We can skip it and use unscanned for now.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh (hunk 4 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: Good call, done.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh (hunk 6 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: Removed this redundant block.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh (hunk 5 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: Uploaded a new checkpoint to `gs://maxtext-model-checkpoints/qwen3-omni-30b-a3b/unscanned/0/items`.

File: tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh (hunk 8 above)
Comment (hengtaoguo) [reply tests/end_to_end/tpu/qwen/moe/qwen3-omni-30b-a3b/2_test_qwen3_omni_30b_a3b.sh]: Thanks for the catch, all base.yml paths have been updated.

**Review by Rohan-Bierneni — APPROVED** [review]
LGTM!
